### ISSUE TITLE
Entity relationship property set to primary key instead of related entity generates error

### ISSUE THREAD

**Issue by vasekbrychta** [nextras/orm-phpstan#4]
Having similar definition
```php
/**
 * @property int     $id          {primary}
 * @property Author  $author      {m:1 Author::$books}
 */
class Book extends Nextras\Orm\Entity\Entity
{}
```

and calling `$book->author = 1;` generates

```
Property Book::$author (Author) does not accept int.
```

although the Author entity is correctly auto-loaded.

**Comment by hrach** [comment]
<del>
Hi, to fix this I would need an actual reproduction. This seems like not properly initialized relationship container. This behaviour is currently [covered](https://github.com/nextras/orm/blob/3cddad369e8df740525e95b1d7a20b5de44a5fee/tests/cases/integration/Relationships/relationships.manyHasOne.phpt#L74) by tests, so there will be some mistake.
</del>

I'm missed the most important thing this is reported on phpstan extension repo... 🤕 . Ok, sorry.

This is known limitation and I'm working on a [fix](https://github.com/nextras/orm-phpstan/tree/entity_relationships). The fix require [some changes in PhpStan](https://github.com/phpstan/phpstan/pull/1962), which I hope are quite near mering

**Comment by hrach** [comment]
In my projects I currently ignore this errors by 
```
- '#Property [\w\\]+::\$\w+ \([\w|\\]+\) does not accept int\.#'
- '#Property [\w\\]+::\$\w+ \([\w|\\]+\) does not accept int|null\.#'
```
Be aware they are not super-save.

**Comment by vasekbrychta** [comment]
Ok, that's great! Thanks for the workaround. I might add the common namespace prefix for all my Entities to be more safe for now.

**Comment by hrach** [comment]
Finally fixed, right now works only with PhpStan master.
